src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -200,11 +200,11 @@ private async Task ProcessQueueAsync()
                     else
                     {
                         // Non mutating are fire-and-forget because they are by definition readonly. Any errors
-                        // will be sent back to the client but we can still capture errors in queue processing
-                        // via NFW, though these errors don't put us into a bad state as far as the rest of the queue goes.
+                        // will be sent back to the client but they can also be captured via HandleNonMutatingRequestError,
+                        // though these errors don't put us into a bad state as far as the rest of the queue goes.
                         // Furthermore we use Task.Run here to protect ourselves against synchronous execution of work
                         // blocking the request queue for longer periods of time (it enforces parallelizabilty).
-                        _ = Task.Run(() => work.StartRequestAsync(context, cancellationToken), cancellationToken);
+                        _ = Task.Run(() => HandleNonMutatingRequestErrorAsync(work.StartRequestAsync(context, cancellationToken)), cancellationToken);
```

Comment:
Should this be trying to use an exception filter somewhere, if able? It's not perfect, obviously, but the downside of this approach is we'll never be able to get dumps at the time of throw, if able.

Reply:
@jasonmalinowski and I talked offline. I filed https://github.com/dotnet/razor-tooling/issues/6950 as a follow-up item for this.